base/glibc-compatibility/musl/getauxval.c

```diff
@@ -90,6 +94,11 @@ static unsigned long NO_SANITIZE_THREAD __auxv_init_procfs(unsigned long type)
     _Static_assert(sizeof(aux) < 4096, "Unexpected sizeof(aux)");
     while (__retry_read(fd, &aux, sizeof(aux)) == sizeof(aux))
     {
+#if defined(__has_feature)
+#if __has_feature(memory_sanitizer)
+        __msan_unpoison(&aux, sizeof(aux));
```

Comment:
This has introduced a new warning:

```
[30/10859] Building C object base/glibc-compatibility/CMakeFiles/glibc-compatibility.dir/musl/getauxval.c.o
/mnt/ch/ClickHouse/base/glibc-compatibility/musl/getauxval.c:99:9: warning: implicit declaration of function '__msan_unpoison' is invalid in C99 [-Wimplicit-function-declaration]
        __msan_unpoison(&aux, sizeof(aux));
```

Reply:
But all builds were ok...

Reply:
> I don't know if this is the desired behaviour for those builds though.

I think it's not, we should either suppress/disable unneeded warnings or turn them into errors

Reply:
I think I was wrong and what's likely happening is that base/ is not covered by -Werror. For the moment I've created a PR to address the warning (https://github.com/ClickHouse/ClickHouse/pull/43065)

Reply:
@Algunenano all ClickHouse code should be covered by -Werror.
Only third-party libraries are exceptions.

Reply:
PR for that here https://github.com/ClickHouse/ClickHouse/pull/43082. It's already detecting issues